flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/MailboxProcessor.java

```diff
@@ -183,27 +198,34 @@ public void runMailboxLoop() throws Exception {
 
         final MailboxController defaultActionContext = new MailboxController(this);
 
-        while (isMailboxLoopRunning()) {
+        while (isNotSuspended()) {
             // The blocking `processMail` call will not return until default action is available.
             processMail(localMailbox, false);
-            if (isMailboxLoopRunning()) {
+            if (isNotSuspended()) {
```

Comment:
nit: I think code would be a bit more self-explanatory if it was

```
while (isMailboxLoopRunning()) {
 ...
 if (isMailboxLoopRunning()) {
```
as it was before, but with modified `isMailboxLoopRunning()` definition to: `mailboxLoopRunning && !suspended`?
 
Maybe if changing the semantic of `isMailboxLoopRunning()` is causing some issues, we could introduce `isMailboxLoopRunningAndNotSuspended()`? Because saying that "mailbox is suspended" when it's actually closed is confusing.

Reply:
I agree that it will be better to have isLoopRunning(or similar). Maybe we indeed rename the old mailboxLoopRunning to mailboxActive(along with method). And new suspended variable rename to loopRunning(or mailboxLoopRunning)?